catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/util/EndpointUtilTest.java

```diff
@@ -0,0 +1,204 @@
+
+    CatalogFramework catalogFrameworkMock;
+
+    FilterBuilder filterBuilderMock;
+
+    List<InjectableAttribute> injectableAttributeList;
+
+    InjectableAttribute injectableAttributeMock;
+
+    AttributeRegistry attributeRegistryMock;
+
+    Filter filterMock;
+
+    AttributeBuilder attributeBuilderMock;
+
+    ContextualExpressionBuilder contextualExpressionBuilderMock;
+
+    QueryResponse responseMock;
+
+    Metacard metacardMock;
+
+    Result resultMock;
+
+    @Before
+    public void setUp() throws Exception {
+
+        // mocks
+        metacardTypeList = new ArrayList<>();
+        metacardTypeMock = mock(MetacardType.class);
+        metacardTypeList.add(metacardTypeMock);
+
+        catalogFrameworkMock = mock(CatalogFramework.class);
+
+        filterBuilderMock = mock(FilterBuilder.class);
+
+        injectableAttributeList = new ArrayList<>();
+        injectableAttributeMock = mock(InjectableAttribute.class);
+        injectableAttributeList.add(injectableAttributeMock);
+
+        attributeRegistryMock = mock(AttributeRegistry.class);
+        filterMock = mock(Filter.class);
+        attributeBuilderMock = mock(AttributeBuilder.class);
+        contextualExpressionBuilderMock = mock(ContextualExpressionBuilder.class);
+
+        responseMock = mock(QueryResponse.class);
+        metacardMock = mock(Metacard.class);
+        resultMock = mock(Result.class);
+
+        // when
+        when(filterBuilderMock.attribute(any())).thenReturn(attributeBuilderMock);
+        when(attributeBuilderMock.is()).thenReturn(attributeBuilderMock);
+        when(attributeBuilderMock.like()).thenReturn(contextualExpressionBuilderMock);
+        when(contextualExpressionBuilderMock.text(anyString())).thenReturn(filterMock);
+        when(catalogFrameworkMock.query(any(QueryRequestImpl.class))).thenReturn(responseMock);
+
+        when(resultMock.getMetacard()).thenReturn(metacardMock);
+        when(metacardMock.getId()).thenReturn("MOCK METACARD");
+
+        // constructor
+        endpointUtil = new EndpointUtil(metacardTypeList,
+                catalogFrameworkMock,
+                filterBuilderMock,
+                injectableAttributeList,
+                attributeRegistryMock);
+
+    }
+
+    @Test
+    public void testGetMetacardsByFilterExpectAll()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+
+        String tagFilter = "";
+        int numResults = 100;
+
+        List<Result> resultList = populateResultMockList(numResults);
+        List<Result> emptyList = populateResultMockList(0);
+
+        when(responseMock.getResults()).thenReturn(resultList, resultList, emptyList);
+
+        Map<String, Result> expected = endpointUtil.getMetacardsByFilter(tagFilter);
+
+        assertThat("Should return " + numResults + " results, but returned " + expected.size(),
+                expected.size() == numResults);
+
+        verify(responseMock, times(4)).getResults();
```

Comment:
❗️ We need to assert that the `ResultIterable` was created with the proper `QueryResquestImpl` object (right start index, page size, etc.) and that the `Map` returned contains all the metacard IDs and `Result` objects returned.
To make this test easier, we should consider shrinking the size of `resultList` to just a few results so we can easily assert that all of them made it in the map.